### PR TITLE
[Docs] Japanese translation of docs/serial_driver.md

### PR DESCRIPTION
**Pull request by umi-umi**
## Description

Add Japanese translation of docs/serial_drive.md.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [x] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [ ] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [ ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by mtei** [comment]
@shelaf @s-show @hsgw @satt99 @takashiski @yoshihara よかったらレビューお願いします。

File: docs/ja/serial_driver.md [hunk 1]
```diff
@@ -0,0 +1,75 @@
+# 'serial' ドライバ
```
Comment (shelaf) [inline docs/ja/serial_driver.md:1]: 他は訳しているのでここも訳していいと思います。
```suggestion
# 'シリアル' ドライバ
```

File: docs/ja/serial_driver.md [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+
+<!---
+  original document: 0.9.51:docs/serial_drive.md
+  git diff 0.9.51 HEAD -- docs/serial_drive.md | cat
+-->
+
+このドライバは[分割キーボード](ja/feature_split_keyboard.md) 機能に力を与えます。
+
+?> この文章でのシリアルは、UART/USART/RS485/RS232 規格の実装ではなく、**一度に1ビットの情報の送信**として読まれるべきです。
```
Comment (shelaf) [inline docs/ja/serial_driver.md:10]:
```suggestion
?> この文章でのシリアルは、UART/USART/RS485/RS232 規格の実装ではなく、**一度に1ビットの情報を送信するもの**として読んでください。
```

File: docs/ja/serial_driver.md [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+                                   //  3: 約 39kbps
+                                   //  4: 約 26kbps
+                                   //  5: 約 20kbps
+```
+
+#### ARM
+
+!> bitbang ドライバは bitbang WS2812 ドライバと接続の問題があります
+
+上記の一般的なオプションに加えて、halconf.h の `PAL_USE_CALLBACKS` 機能もオンにする必要があります。
```
Comment (shelaf) [inline docs/ja/serial_driver.md:48]: in your halconf.hとあるように、必要に応じてキーボードごとに作るhalconf.hでオンにする必要があります。
```suggestion
上記の一般的なオプションに加えて、halconf.h で `PAL_USE_CALLBACKS` 機能もオンにする必要があります。
```

File: docs/ja/serial_driver.md [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+
+### USART Half-duplex
+通信が USART ハードウェアデバイスにオフロードされる STM32 ボードが対象。これにより高速で正確なタイミングを提供されることが利点です。このドライバの `SOFT_SERIAL_PIN` は、設定された USART TX ピンです。**TX ピンに適切なプルアップ抵抗が必要です**。設定するには、以下を rules.mk に追加します:
```
Comment (shelaf) [inline docs/ja/serial_driver.md:51]:
```suggestion
通信が USART ハードウェアデバイスに送信される STM32 ボードが対象です。これにより高速で正確なタイミングを提供できることが利点です。このドライバの `SOFT_SERIAL_PIN` は、設定された USART TX ピンです。**TX ピンに適切なプルアップ抵抗が必要です**。設定するには、以下を rules.mk に追加します:
```

File: docs/ja/serial_driver.md [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+
+```make
+SERIAL_DRIVER = usart
+```
+
+config.h を介してハードウェアを設定します:
+```c
+#define SOFT_SERIAL_PIN B6  // USART TX ピン
+#define SELECT_SOFT_SERIAL_SPEED 1 // or 0, 2, 3, 4, 5
```
Comment (shelaf) [inline docs/ja/serial_driver.md:60]:
```suggestion
#define SELECT_SOFT_SERIAL_SPEED 1 // または 0, 2, 3, 4, 5
```

File: docs/ja/serial_driver.md [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+                                   //  0: 約 460800 ボー
+                                   //  1: 約 230400 ボー (デフォルト)
+                                   //  2: 約 115200 ボー
+                                   //  3: 約 57600 ボー
+                                   //  4: 約 38400 ボー
+                                   //  5: 約 19200 ボー
+#define SERIAL_USART_DRIVER SD1 // TX ピンの USART ドライバ。デフォルトは SD1
+#define SERIAL_USART_TX_PAL_MODE 7 // "代替関数" ピン。ご使用の MCU の適切な値については、それぞれのデータシートを見てください。デフォルトは 7
```
Comment (shelaf) [inline docs/ja/serial_driver.md:68]:
```suggestion
#define SERIAL_USART_TX_PAL_MODE 7 // 「代替機能」 ピン。MCU の適切な値については、それぞれのデータシートを見てください。デフォルトは 7
```

File: docs/ja/serial_driver.md [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+```
+
+また、ChibiOS `SERIAL` 機能を有効にする必要があります:
+* ボードの halconf.h: `#define HAL_USE_SERIAL TRUE`
```
Comment (shelaf) [inline docs/ja/serial_driver.md:72]:
```suggestion
* キーボードの halconf.h: `#define HAL_USE_SERIAL TRUE`
```

File: docs/ja/serial_driver.md [hunk 8]
```diff
@@ -0,0 +1,75 @@
...
+* ボードの mcuconf.h: `#define STM32_SERIAL_USE_USARTn TRUE` (ここで、'n' は MCU で選択した USART のペリフェラル番号と一致)
```
Comment (shelaf) [inline docs/ja/serial_driver.md:73]:
```suggestion
* キーボードの mcuconf.h: `#define STM32_SERIAL_USE_USARTn TRUE` (ここで、'n' は MCU で選択した USART のペリフェラル番号と一致)
```

File: docs/ja/serial_driver.md [hunk 9]
```diff
@@ -0,0 +1,75 @@
...
+
+このドライバは[分割キーボード](ja/feature_split_keyboard.md) 機能に力を与えます。
+
+?> この文章でのシリアルは、UART/USART/RS485/RS232 規格の実装ではなく、**一度に1ビットの情報の送信**として読まれるべきです。
+
+このカテゴリの全てのドライバには以下の特徴があります:
+* 1本の線上でデータと信号を提供
+* シングルマスタ、シングルスレーブに限定
+
+## サポートされるドライバの型
```
Comment (shelaf) [inline docs/ja/serial_driver.md:16]: 形式とか種類とかですかねぇ
```suggestion
## サポートされるドライバの種類
```

**Review by s-show — APPROVED** [review]

**Review by shelaf — APPROVED** [review]

File: docs/ja/serial_driver.md [hunk 10]
```diff
@@ -0,0 +1,75 @@
+# 'シリアル' ドライバ
+
+<!---
+  original document: 0.9.51:docs/serial_drive.md
+  git diff 0.9.51 HEAD -- docs/serial_drive.md | cat
+-->
+
+このドライバは[分割キーボード](ja/feature_split_keyboard.md) 機能を強化します。
```
[the remark this answers is not in the document]
Comment (umi-umi) [reply docs/ja/serial_driver.md]: ご指摘ありがとうございます。
power up 的なとらえ方(増強)をしていたのですが、その後の内容的にも power on (オン/オフ)のほうだったんですね。ありがとうございます。

**Review by mtei — APPROVED** [review]

**Comment by mtei** [comment]
I think this is ready to merge.

**Review by noroadsleft — APPROVED** [review]

**Comment by noroadsleft** [comment]
ありがとうございました！

**Pull request merged**